File: Frontend/types/report.ts
export interface TrendingMovieReport {
  MID: number;
  MName: string;
  Description: string | null;
}

export interface PerformanceLogRecord {
  MID: number;
  MName: string;
  ShowDate: string;
  TCOUNT: number;
  INCOME: number | string;
}

export interface PerformanceLogPage {
  items: PerformanceLogRecord[];
  total: number;
  page: number;
  limit: number;
}

export interface PerformanceLogParams {
  search?: string;
  start_date?: string;
  end_date?: string;
  page?: number;
  limit?: number;
}

File: Backend/schemas/report.py
from datetime import date
from decimal import Decimal
from pydantic import BaseModel


class TrendingMovieReportResponse(BaseModel):
    MID: int
    MName: str
    Description: str | None = None


class PerformanceLogResponse(BaseModel):
    MID: int
    MName: str
    ShowDate: date
    TCOUNT: int
    INCOME: Decimal


class PerformanceLogPageResponse(BaseModel):
    items: list[PerformanceLogResponse]
    total: int
    page: int
    limit: int
